test/dummy/app/tasks/maintenance/custom_enumerating_task.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+module Maintenance
+  class CustomEnumeratingTask < MaintenanceTasks::Task
```

Comment:
I created this for use in tests, but I wasn't able to figure out where the test would belong. It seems like a lot of the functionality of `Task` doesn't have unit tests, and is instead "integration" tested by actually running a task. 🤔

Reply:
Yeah, I think in keeping with how we test our other Task "types", I'd write a test `.perform_now can use custom enumerator instead of collection` in `task_job_test.rb` for now to act as our integration test (it can go with `test '.perform_now accepts Active Record Relations as collection'` and `test '.perform_now accepts CSVs as collection'`), and have it run this new Task. We could write some unit tests for the new enumeration building logic that's been added to `task.rb`, but it is all tested through `task_job_test.rb`, and since it's all private API I'd be okay to just continue relying on the tests in TaskJob to exercise that functionality.

Reply:
`Task#enumerator_builder` is `public` though, and definitely part of the public API if we're telling consumers to override it to implement custom tasks. I'm not sure if that changes anything, since the unit tests for the other public methods are pretty lightweight.

Reply:
Ah true,`Task#enumerator_builder` is indeed public now. And it does have branching, so that could be an indication that we should test it (even if it's implicitly tested in `TaskJob`). The unit tests for the other methods are mostly laying out the template for a Task, which is why they're so lightweight.

Anyways, I'll leave it up to you, IMHO it comes down to personal preference mostly 😅  We have a test for each collection type, and another for the error scenario all in `TaskJobTest`, so I consider it to have test coverage, but totally up to you!